jdaviz/configs/cubeviz/plugins/viewers.py

```diff
@@ -144,46 +142,62 @@ def _get_marks_for_layers(self, layers):
 
     def _on_subset_delete(self, msg):
         # delete any ShadowSpatialSpectral mark for which either of the spectral or spatial marks
-        # no longer exists
-        spectral_marks = self._get_marks_for_layers(self._get_spectral_subset_layers())
-        spatial_marks = self._get_marks_for_layers(self._get_spatial_subset_layers())
+        # no longer exists by matching the uuid of the msg subset to the uuid of the subsets
+        # in ShadowSpatialSpectral
+        spectral_marks = self._get_marks_for_layers([layer for layer in self._get_spectral_subset_layers() # noqa
+                                                     if layer.layer.uuid == msg.subset.uuid])
+        spatial_marks = self._get_marks_for_layers([layer for layer in self._get_spatial_subset_layers() # noqa
+                                                    if layer.layer.uuid == msg.subset.uuid])
         self.figure.marks = [m for m in self.figure.marks
-                             if not (isinstance(m, ShadowSpatialSpectral) and
-                                     (m.spatial_spectrum_mark in spatial_marks or
-                                      m.spectral_subset_mark in spectral_marks))]
+                             if not (isinstance(m, ShadowSpatialSpectral)
+                                     and msg.subset.uuid in [m.spatial_uuid, m.spectral_uuid])]
 
     def _expected_subset_layer_default(self, layer_state):
         """
         This gets called whenever the layer of an expected new subset is added, we want to set the
         default for the linewidth depending on whether it is spatial or spectral, and handle
         creating any necessary marks for spatial-spectral subset intersections.
         """
+        # TODO: this does not seem to be called correctly/in the right order when plugin data
+        #  is added AFTER subsets
         super()._expected_subset_layer_default(layer_state)
-
         subset_type = get_subset_type(layer_state.layer)
         if subset_type is None:
             return
 
         this_mark = self._get_marks_for_layers([layer_state])[0]
 
         new_marks = []
-
         if subset_type == 'spatial':
             layer_state.linewidth = 1
 
             # need to add marks for every intersection between THIS spatial subset and ALL spectral
-            spectral_marks = self._get_marks_for_layers(self._get_spectral_subset_layers())
-            for spectral_mark in spectral_marks:
-                new_marks += [ShadowSpatialSpectral(this_mark, spectral_mark)]
-
-        else:
+            # spectral_marks = self._get_marks_for_layers(self._get_spectral_subset_layers())
+            spectral_layers = [sub_layer for sub_layer in self._get_spectral_subset_layers()
+                               if sub_layer.layer.data.label == layer_state.layer.data.label]
+            spectral_marks = self._get_marks_for_layers(spectral_layers)
```

Comment:
I think this is cause the problem mentioned in your comment above.  If duplicate marks were being added before, maybe we just need some logic somewhere to prevent those from being added, but I don't think we want to limit to just this spectral subset at this point in the logic 🤔